celery/worker/consumer/consumer.py

```diff
@@ -448,18 +449,42 @@ def on_close(self):
         # They can't be acked anyway, as a delivery tag is specific
         # to the current channel.
         if self.controller and self.controller.semaphore:
-            self.controller.semaphore.clear()
+            try:
+                self.controller.semaphore.clear()
+            except Exception as exc:
+                error('Error clearing semaphore: %r', exc, exc_info=True)
```

Comment:
How does help this and below logs?

Reply:
This function is a single point where the worker connection with the message broker gets closed. So, we are logging the errors, if any, that occurred along the connection closure step that prevented releasing this connection.